### PR TITLE
Domains: New inbound transfer domain status design

### PR DESCRIPTION
**Pull request by eltongo**
#### Changes proposed in this Pull Request

This PR introduces the new design for inbound transfer domains.

Transfer has not started yet:

Screenshot 2020-03-27 at 12 27 02 PM

Transfer in progress:

Screenshot 2020-03-27 at 12 29 19 PM

Transfer failed:

Screenshot 2020-03-27 at 12 32 01 PM

#### Testing instructions

Try to transfer a domain (or by hacking the backend), and ensure that the proper view is shown based on the domain's transfer status.

NOTE: The pending status (first screenshot above) is shown when transfer is purchased from NUX

### CODE REVIEW

File: client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+		) {
+			return null;
+		}
+
+		let title;
+
+		if ( domain.transferStatus === transferStatus.CANCELLED ) {
+			title = translate( 'Remove transfer from your account' );
+		} else {
+			title = translate( 'Cancel transfer' );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **26** times:
`translate( 'Cancel Transfer' )` *ES Score: 12*

File: client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+		if ( status === transferStatus.PENDING_START ) {
+			return {
+				statusText: translate( 'Action required' ),
+				statusClass: 'status-error',
+				icon: 'info',
+			};
+		} else if ( status === transferStatus.CANCELLED ) {
+			return {
+				statusText: translate( 'Transfer failed' ),
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Transfer Failed' )` *ES Score: 11*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-status-inbound-transfer)

File: client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+				statusClass: 'status-error',
+				icon: 'info',
+			};
+		}
+
+		return {
+			statusText: translate( 'Transfer in progress' ),
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Transfer in Progress' )` *ES Score: 10*

File: client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+						{
+							args: {
+								domain: domain.name,
+							},
+						}
+					) }
+				</p>
+
+				<Button primary onClick={ this.startTransfer }>
+					{ translate( 'Start transfer' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Start Transfer' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-status-inbound-transfer)

File: client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+		);
+	}
+
+	renderTransferInProgress() {
+		const { translate } = this.props;
+
+		return (
+			<p>
+				{ translate(
+					'Your transfer has been started and is waiting for authorization from your current ' +
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'This transfer has been started and is waiting for authorization from your current provider. If you need to cancel the transfer, please contact them for assistance.' )` *ES Score: 6*

File: client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+										target="_blank"
+										rel="noopener noreferrer"
+									/>
+								),
+							},
+						}
+					) }
+				</p>
+
+				<Button onClick={ this.startTransfer }>{ translate( 'Start transfer again' ) }</Button>
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/domain-types/transfer-in-domain-type.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Start Transfer Again' )` *ES Score: 10*

File: client/my-sites/domains/domain-management/edit/navigation/index.jsx [hunk 7]
```diff
@@ -251,10 +251,25 @@ class DomainManagementNavigation extends React.Component {
...
+			const status = domain.transferStatus;
+
+			if ( transferStatus.PENDING_OWNER === status || transferStatus.PENDING_REGISTRY === status ) {
+				return null;
+			}
+
+			if ( status === transferStatus.CANCELLED ) {
+				title = translate( 'Remove transfer from your account' );
+			} else {
+				title = translate( 'Cancel transfer' );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/navigation/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **26** times:
`translate( 'Cancel Transfer' )` *ES Score: 12*

**Comment by hambai** [comment]
> I tested this with domains transferring in and it all looked good. I can't get a failed state to trigger but wanted to mention we should use red for the highlight since it's a failed action.

Hey @fditrapani - I was under the impression that the pink status is our "error" status. The pink and the red are really close to each other (to be honest I never noticed any difference while looking at the Figma screens) - are we sure we want to introduce two similar colours if they have so similar meaning? Also if we want to have pink and red - should we set to red some of the other domain statuses - like expired domain for example seems like a good candidate to have even higher priority?

**Comment by fditrapani** [comment]
> I was under the impression that the pink status is our "error" status. The pink and the red are really close to each other (to be honest I never noticed any difference while looking at the Figma screens) - are we sure we want to introduce two similar colours if they have so similar meaning? 

This is a good point and something I definitely want to revisit. The way I was using the pink, our primary colour, was to encourage action vs. to induce alarm or anxiety. The red is there for complete failed state where no action can be taken. These colours of course, aren't playing well with our colour schemes and hence another reason to revisit.

> Also if we want to have pink and red - should we set to red some of the other domain statuses - like expired domain for example seems like a good candidate to have even higher priority?

Yes, this is definitely something worth considering. Maybe we should turn all the pinks to reds (we can deal with this in a separate PR). Anyways, how about we go forward with the colours as is and we can revisit later then?

**Pull request merged**